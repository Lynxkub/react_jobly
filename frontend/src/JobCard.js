import React , {useState , useEffect} from 'react';
import useToggleApplyState from './hooks/useToggleApplyState';


const JobCard = ({job}) => {
  
    const [isApplied , toggleIsApplied] = useToggleApplyState()
    const [clicked , setClicked] = useState(false)

    useEffect(() => {
        function switchCheck() {
        const jobsAppliedTo = JSON.parse(localStorage.getItem('jobs'));
        const checkApplied = jobsAppliedTo.find(j => j === `${job.id}`);
       
       
        if(checkApplied === `${job.id}`){
            setClicked(true)
        }
        }
        try{
        switchCheck();
        }catch (e){
            console.log(e);
        }
    } , [])
 
    
    return(
        <div>
        
           <div className ='card bg-dark' style={{'width': '18rem'}}>
               <div className='card-body'>
                    <h5 className='card-title text-light'>{job.title}</h5>
                    <h6 className='card-subtitle mb-2 text-light'>{job.companyName}</h6>
                    <h6 className='card-subtitle mb-2 text-light'>Salary : {job.salary}</h6>
                    <h6 className='card-subtitle mb-2 text-light'>Equity : {job.equity ? job.equity : '0'}</h6>
                    <button type='button' className = {!isApplied && !clicked ? 'btn btn-success btn-sm' : 'btn btn-danger btn-sm'} onClick={toggleIsApplied} id = {job.id}>{!isApplied &&!clicked ? 'Apply' : 'Remove Application'}</button>
                    
                    
                    </div>
                    </div>
        </div>
    )
}

export default JobCard;

