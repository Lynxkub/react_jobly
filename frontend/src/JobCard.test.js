import React from 'react';
import {render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import JobCard from './JobCard';



it('renders without crashing' , () => {
    const job = {
        id : 1 ,
        title : 'test',
        salary : 100,
        equity : '0'
    }
    render(<JobCard job = {job} />)
})

it('matches snapshot' , () => {
    const job = {
        id : 1 ,
        title : 'test',
        salary : 100,
        equity : '0'
    }

    const {asFragment} = render(<JobCard job = {job} />)
    expect(asFragment()).toMatchSnapshot();
})