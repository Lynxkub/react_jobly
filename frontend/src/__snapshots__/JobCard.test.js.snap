// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h2>
        test
      </h2>
      <h2 />
      <h3>
        100
      </h3>
      <h3>
        0
      </h3>
      <button
        id="1"
      >
        Apply
      </button>
    </div>
  </div>
</DocumentFragment>
`;
